### PR TITLE
feat: add in scheduling settings to settings app [DHIS2-15765]

### PR DESCRIPTION
**Pull request by tomzemp**
See https://dhis2.atlassian.net/jira/software/c/projects/DHIS2/issues/DHIS2-15765

This adds scheduling settings into the settings app (in a new section called "Scheduled jobs" (approved by Joe on ticket)):

**After**
image

Note: the front-end validations do not appear to generally be working in the settings app, and the backend does not currently have validation of provided values [I guess we could fix the front end, but that seems like a separate ticket]

@jbee : I've tried to summarize your descriptions into shorter text for the purpose of presenting to the user in the settings app, but let me know if you think anything should be changed.

### CODE REVIEW

File: src/settingsKeyMapping.js [hunk 1]
```diff
@@ -689,7 +689,49 @@ const settingsKeyMapping = {
...
+        ),
+        type: 'textfield',
+        inputType: 'number',
+        minValue: 1,
+        maxValue: 24,
+        validators: ['positive_number'],
+    },
+    jobsLogDebugBelowSeconds: {
+        label: i18n.t(
+            'Job execution interval (seconds) below which a job will be logged at debug (rather than info) level (20+)'
```
Comment (jbee) [inline src/settingsKeyMapping.js:727]: There seems to be no message in the `en.pot` file for this one. No sure this is required but I would have expected it given the other new settings have an entry there.

Comment (tomzemp) [reply src/settingsKeyMapping.js:727]: This file's auto-generated, and it will be re-generated on the next updated

File: i18n/en.pot [hunk 2]
```diff
@@ -817,3 +754,16 @@ msgstr "Remote server username"
 
 msgid "Remote server password"
 msgstr "Remote server password"
+
+msgid "Number of minutes after which a job is considered stale"
```
[the remark this answers is not in the document]
Comment (tomzemp) [reply i18n/en.pot:758]: Number of minutes after which a _stale_ job is _reset to scheduled state_
@jbee ?

File: i18n/en.pot [hunk 3]
```diff
@@ -817,3 +754,16 @@ msgstr "Remote server username"
...
+msgstr "Number of minutes after which a job is considered stale"
+
+msgid "Number of minutes after which a completed one-run job is deleted"
```
[the remark this answers is not in the document]
Comment (tomzemp) [reply i18n/en.pot:761]: Number of minutes after which a completed _ad-hoc_ job is deleted
@jbee ?

File: i18n/en.pot [hunk 4]
```diff
@@ -817,3 +754,16 @@ msgstr "Remote server username"
...
+msgstr "Number of minutes after which a completed one-run job is deleted"
+
+msgid ""
+"Number of hours a job will trigger after its intended time if job has not "
+"yet run"
```
[the remark this answers is not in the document]
Comment (tomzemp) [reply i18n/en.pot:766]: _Maximum_ number of hours a job _may_ trigger after its intended time if job has not yet run
@jbee ?

**Comment by tomzemp** [comment]
@jbee : thanks for the comments. I've made some small edits based on your comments, but feel free to suggest something else. I think we just need not overly long descriptions for within the app, but I assume that what these settings do will also be explained more fully in the documentation?

File: src/form-fields/text-field.js [hunk 5]
```diff
@@ -56,6 +56,7 @@ class TextFieldComponent extends React.Component {
                     {...other}
                     value={this.state.value}
                     onChange={this.onChange}
+                    error={true}
```
Comment (Mohammer5) [inline src/form-fields/text-field.js:59]: This change is hard to follow. Could this introduce regressions? I appears as if this would be applied to all text fields, but I can't see it being applied.

Regardless of that, you can just do `error` instead of `error={true}`

Comment (tomzemp) [reply src/form-fields/text-field.js]: 😱 Yikes. No; that change is not supposed to be there. I was beginning to try to look at why validation errors weren't showing up, but I guess I made this debugging change before switching to a new branch.

Removed now. Thanks!

**Review by Mohammer5 — APPROVED** [review]
